cmake/spdlog.cmake

```diff
@@ -1,33 +1,24 @@
 include(FetchContent)
 
 message(STATUS "Setting up spdlog")
+set(SPDLOG_VERSION 1.14.1)
 
-find_package(spdlog 1.13.0 QUIET)
-
-if (${spdlog_FOUND})
-
-    message(STATUS "Using existing spdlog installation")
+# Known Issue:
+# If you install spdlog@1.14.1 via homebrew on ARM macOS, CMake will find spdlog
+# However, there is a version mismatch between the `fmt` library installed as dependency, and the one actually
+# being required leading to a linking error (i.e. missing symbols) while compiling!
```

Comment:
should this be in a comment here? Not saying it shouldn't be just wondering where the best place is to put something like this? I don't cmake has try catch, does it? :D

Reply:
This should probably not even belong in our repo; I am just nice if other people try to compile on macOS and use `spdlog` installed from `brew`.

I'll remove it when they upgrade the version.